### ISSUE TITLE
we suggest to run 'gcloud components update' when the sdk is auto-managed

### ISSUE THREAD

**Issue by patflynn** [GoogleCloudPlatform/google-cloud-intellij#2028]
an error deploying showed the following output:

```
Failed to deploy '[2018-04-12 15:13:06] Maven build: hello. Project: positive-karma-844. Version: auto': Deployment failed with exit code: 1
Please make sure that you are using the latest version of the Google Cloud SDK.
Run ''gcloud components update'' to update the SDK. (See: https://cloud.google.com/sdk/gcloud/reference/components/update.)
```

**Comment by ivanporty** [comment]
I guess we need to ditch that error message when managed SDK is used.

**Comment by etanshaul** [comment]
that error was kind of a generic catchall that we showed this user in case we didn't have more information to help them. we can reconsider this though

**Comment by etanshaul** [comment]
when the managed sdk is used that error message would only make sense if automatic updates are not enabled

**Comment by ivanporty** [comment]
Right, we can alternatively suggest go to "Settings->Google->Cloud SDK" and attempt to update manually (using a link) in this case instead of asking to run a command on SDK which location is not even known to users. Or even better, in case of an unknown failure check if SDK is up to date, and if not, show suggestion to update, otherwise if it is up-to-date, then just say 'Failed to deploy'. WDYT?
